pages/api/public/users/reset-password.tsx

```diff
@@ -0,0 +1,47 @@
+import type { NextApiRequest, NextApiResponse } from "next";
+import jwt from "jsonwebtoken";
+import { prisma } from "../../../../lib/prisma";
+import { sendPasswordResetNotifyEmail } from "../../../../lib/email";
+
+export default async function handle(
+  req: NextApiRequest,
+  res: NextApiResponse
+) {
+  // POST /api/public/users/reset-password
+  // Resets a users password
+  // Required fields in body: token, hashedPassword
+  if (req.method === "POST") {
+    const { token, hashedPassword } = req.body;
+
+    try {
+      const { id } = await jwt.decode(token);
```

Comment:
I think you are missing the token verification here to ensure that the token is not expired or has the wrong signature, e.g.
```
const isValid = await new Promise((resolve) => {
            jwt.verify(
              credentials?.token,
              serverRuntimeConfig.nextauthSecret + user.email,
              (err) => {
                if (err) resolve(false);
                if (!err) resolve(true);
              }
            );
          });

          if (!isValid) {
            throw new Error("Token is not valid or expired");
          }
```

Reply:
Great finding! - Thank you!